torch/csrc/jit/passes/python_print.cpp

```diff
@@ -0,0 +1,588 @@
+          printValueList(out, node->outputs(), "", ", = ");
+        }
+        out << useOf(node->input()) << "\n";
+        break;
+      default:
+
+        std::stringstream ss;
+        printRHS(ss, node);
+
+        // this node is safe to inline, so assign the output value
+        // to that expression directly
+        // guard against really long lines
+        if (output_inline_.count(node) > 0 && ss.str().size() + level * 2 < 40) {
+          assignValue(node->output(), ss.str());
+          return;
+        }
+        assignValuesToTheirUniqueNames(node->outputs());
+        indent();
+        // Print outputs
+        if (node->outputs().size() > 0) {
+          printValueList(out, node->outputs());
+          out << " = ";
+        }
+        out << ss.str() << "\n";
+    }
+  }
+
+  size_t addTensorConstant(at::Tensor t) {
+    tensor_constants.emplace_back(std::move(t));
+    return tensor_constants.size() - 1;
+  }
+
+  // Prints the RHS value of a Node, e.g. `aten.add(x, y)`
+  void printRHS(std::ostream& stmt, const Node* node) {
+    switch(node->kind()) {
+      case PythonOp::Kind: {
+        auto value = static_cast<const PythonOp*>(node);
+        if (enforce_importable_) {
+          throw script::ErrorReport(node->getSourceLocation())
+              << "could not export python function call " << value->name()
+              << ". Remove calls to python functions before export.";
+        }
+
+        stmt << "^" << value->name();
+        value->writeScalars(stmt);
+        printValueList(stmt, node->inputs(), "(", ")");
+      } break;
+      case prim::Constant: {
+        IValue v = toIValue(node->output()).value();
+        if(v.isTensor()) {
+          stmt << "CONSTANTS.c" << addTensorConstant(std::move(v).toTensor());
+        } else if(v.isString()) {
+          // TODO: escape the string correctly by implementing a subset of
+          // string escapes in both printing and parsing.
+          stmt << "\"" << v.toStringRef() << "\"";
+        } else if(v.isTensorList()) {
+          auto tl = v.toTensorListRef();
+          stmt << "[";
+          const char* delim = "";
+          for(at::Tensor t : tl) {
+            stmt << delim << "CONSTANTS.c" << addTensorConstant(std::move(t));
+            delim = ", ";
+          }
+          stmt << "]";
+        } else {
+          // TODO: ensure floats always print with their periods
+          stmt << v;
+        }
+      } break;
+      case prim::None:
+      case prim::NoneGenerator:
+      case prim::Undefined: {
+        stmt << "None";
+      } break;
+      case prim::FloatToInt: {
+        printValueList(stmt, node->inputs(), "int(", ")");
+      } break;
+      case prim::StringToFloat:
+      case prim::IntToFloat: {
+        printValueList(stmt, node->inputs(), "float(", ")");
+      } break;
+      case prim::TensorToBool: {
+        printValueList(stmt, node->inputs(), "bool(", ")");
+      } break;
+      case prim::Print: {
```

Comment:
Looks like it's missing prim::RaiseException here

Reply:
RaiseException was accidentally listed in handled, but it doesn't need to be because it has schema. I removed it. ImplicitTensorToNum is similar: it has schema so it doesn't need a special case.